lib/sendgrid/base_interface.rb

```diff
@@ -0,0 +1,36 @@
+require 'ruby_http_client'
+require_relative 'version'
+
+# Initialize the HTTP client
+class BaseInterface
+  attr_accessor :client
+  attr_reader :request_headers, :host, :version, :impersonate_subuser
+  # * *Args* :
+  #   - +auth+ -> authorization header value
+  #   - +host+ -> the base URL for the API
+  #   - +request_headers+ -> any headers that you want to be globally applied
+  #   - +version+ -> the version of the API you wish to access,
+  #                  currently only "v3" is supported
+  #   - +impersonate_subuser+ -> the subuser to impersonate, will be passed
+  #                              in the "On-Behalf-Of" header
+  #
+  def initialize(auth:, host:, request_headers: nil, version: nil, impersonate_subuser: nil)
+    @auth = auth
+    @host = host
+    @version = version ? version : 'v3'
```

Comment:
why even have this field is the only supported value is "v3"?

Reply:
v2 still exists so it would just be `api` is guess: https://sendgrid.com/docs/API_Reference/Web_API/mail.html

Reply:
Not sure I understand. v2 still exists but our helper lib doesn't support it. According to the docs above, `currently only "v3" is supported`. If that's the case, why do we allow the user to pass in their own value?

Reply:
The underlying client doesn't _not_ support it. It's just there are no higher-level helpers to work with v2, but you could still use this client.

Reply:
Ah. Makes sense.